### PR TITLE
Fix #4184 - Wrap "PhotovoltaicPerformance:Sandia" in OS SDK

### LINKED ISSUE #4184: Provide OS SDK / API Access to EnergyPlus "PhotovoltaicPerformance:Sandia" object
Enhancement Request
--------------
Provide OS SDK / API Access to EnergyPlus "PhotovoltaicPerformance:Sandia" object 

## Detailed Description
This enhancement would allow improved accuracy of OS measure based simulation workflows that combine on-site PV generation with or without battery storage that are used to examine envelope and other HVAC related thermal load reduction measures.  As an example, users need to be able to dynamically evaluate the impact-related trade-offs of specific envelope improvements (which reduce peak HVAC loads/equip size, which reduce peak HVAC equipment electricity demand) against each other. Currently, an EnergyPlus measure can be authored to allow access to the EnergyPlus "PhotovoltaicPerformance:Sandia" object, but this non-standard approach makes it difficult and fragile to design scalable OS measure based workflows for evaluating different on site generation options.

## Possible Implementation
I understand that NREL SAM features (including improved PV and battery storage modeling) are being integrated into the next release of EnergyPlus (and I hope that OpenStudio team wraps those new E+ features/capabilities as well). Nevertheless, at first glance a 1:1 wrapping of the "PhotovoltaicPerformance:Sandia" object and it's attributes should be suitable for OS users to access via the API/SDK.

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

- Fix #4184
-  Wrap "PhotovoltaicPerformance:Sandia" in OS SDK

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/energyplus/ForwardTranslator.cpp [hunk 1]
```diff
@@ -3196,6 +3201,19 @@ namespace energyplus {
     // result.push_back(IddObjectType::OS_PhotovoltaicPerformance_EquivalentOneDiode);
     // result.push_back(IddObjectType::OS_PhotovoltaicPerformance_Simple);
 
+    // TODO: These should be translated by their ElectricLoadCenter_Distribution, shouldn't they?
```
Comment (kbenne) [inline src/energyplus/ForwardTranslator.cpp:3204]: I think yes. especially if these objects do not function on their own. (without the load center)

**Review by kbenne — CHANGES_REQUESTED** [review]
@jmarrec. I updated the branch and resolved a conflict with develop. And I had one comment about your "TODO" related to the catch all translate. If you can take a look at that and either remove the TODO comment or remove those objects from the list altogether, then I think this can go after CI is done.

File: src/energyplus/ForwardTranslator.cpp (hunk 1 above)
Comment (jmarrec) [reply src/energyplus/ForwardTranslator.cpp]: Cleaned merge conflict in https://github.com/NREL/OpenStudio/pull/4194/commits/8268d311234f4a6a71aab4f9d0c860dac2d567b2

File: src/energyplus/ForwardTranslator.cpp [hunk 2]
```diff
@@ -3192,9 +3197,10 @@ namespace energyplus {
     // result.push_back(IddObjectType::OS_ElectricLoadCenter_Storage_Simple);
     // result.push_back(IddObjectType::OS_ElectricLoadCenter_Storage_Converter);
 
-    // Generator_Photovoltaic is responsible for translating these two
+    // Generator_Photovoltaic is responsible for translating these three
     // result.push_back(IddObjectType::OS_PhotovoltaicPerformance_EquivalentOneDiode);
     // result.push_back(IddObjectType::OS_PhotovoltaicPerformance_Simple);
+    // result.push_back(IddObjectType::OS_PhotovoltaicPerformance_Sandia);
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator.cpp:3203]: @kbenne see diff is clean now.

**Pull request merged**